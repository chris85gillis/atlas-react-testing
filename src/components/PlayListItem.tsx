import React from 'react';

interface PlayListItemProps {
  title: string;
  genre: string;
  length: string;
  onClick?: () => void;
}

const PlayListItem: React.FC<PlayListItemProps> = ({ title, genre, length }) => {
  return (
    <div className="flex justify-between items-center py-2 flex-nowrap p-6">
      <div className="flex-1 min-w-0">
        <h2 className="text-lg font-semibold text-nightText truncate">{title}</h2>
        <p className="text-sm text-nightSecondary truncate">{genre}</p>
      </div>
      <span className="text-sm text-nightSecondary whitespace-nowrap">{length}</span>
    </div>
  );
};

export default PlayListItem;