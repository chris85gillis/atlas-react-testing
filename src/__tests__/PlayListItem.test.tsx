import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';

describe('PlayListItem', () => {
  it('renders correctly with all props', () => {
    const { asFragment } = render(<PlayListItem title="Song Title" genre="Pop" length="3:45" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with missing genre', () => {
    const { asFragment } = render(<PlayListItem title="Song Title" genre="" length="3:45" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with missing length', () => {
    const { asFragment } = render(<PlayListItem title="Song Title" genre="Pop" length="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});