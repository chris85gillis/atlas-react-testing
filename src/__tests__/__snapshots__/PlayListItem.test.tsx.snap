// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem > renders correctly with all props 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center py-2 flex-nowrap p-6"
  >
    <div
      class="flex-1 min-w-0"
    >
      <h2
        class="text-lg font-semibold text-nightText truncate"
      >
        Song Title
      </h2>
      <p
        class="text-sm text-nightSecondary truncate"
      >
        Pop
      </p>
    </div>
    <span
      class="text-sm text-nightSecondary whitespace-nowrap"
    >
      3:45
    </span>
  </div>
</DocumentFragment>
`;

exports[`PlayListItem > renders correctly with missing genre 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center py-2 flex-nowrap p-6"
  >
    <div
      class="flex-1 min-w-0"
    >
      <h2
        class="text-lg font-semibold text-nightText truncate"
      >
        Song Title
      </h2>
      <p
        class="text-sm text-nightSecondary truncate"
      />
    </div>
    <span
      class="text-sm text-nightSecondary whitespace-nowrap"
    >
      3:45
    </span>
  </div>
</DocumentFragment>
`;

exports[`PlayListItem > renders correctly with missing length 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center py-2 flex-nowrap p-6"
  >
    <div
      class="flex-1 min-w-0"
    >
      <h2
        class="text-lg font-semibold text-nightText truncate"
      >
        Song Title
      </h2>
      <p
        class="text-sm text-nightSecondary truncate"
      >
        Pop
      </p>
    </div>
    <span
      class="text-sm text-nightSecondary whitespace-nowrap"
    />
  </div>
</DocumentFragment>
`;
